tensorboard/webapp/metrics/store/metrics_selectors_test.ts

```diff
@@ -425,45 +426,172 @@ describe('metrics selectors', () => {
-        buildMetricsState({
+    describe('when linked time is disabled', () => {
+      let partialState: Partial<MetricsState>;
+      beforeEach(() => {
+        partialState = {
           linkedTimeEnabled: false,
-          cardStateMap: {
-            card1: {
-              timeSelection: {
-                start: {step: 0},
-                end: {step: 5},
+        };
+      });
+
+      it('returns undefined if step selection is disabled', () => {
+        expect(
+          selectors.getMetricsCardTimeSelection(
+            appStateFromMetricsState(
+              buildMetricsState({
+                ...partialState,
+                stepSelectorEnabled: true,
+                cardStateMap: {
+                  card1: {
+                    stepSelection: CardSelectionState.DISABLED,
+                    dataMinMax: {
+                      minStep: 0,
+                      maxStep: 10,
+                    },
+                    timeSelection: {
+                      start: {step: 0},
+                      end: {step: 5},
+                    },
+                  },
+                },
+              })
+            ),
+            'card1'
+          )
+        ).toBeUndefined();
+
+        expect(
+          selectors.getMetricsCardTimeSelection(
+            appStateFromMetricsState(
+              buildMetricsState({
+                ...partialState,
+                stepSelectorEnabled: false,
+                cardStateMap: {
+                  card1: {
+                    dataMinMax: {
+                      minStep: 0,
+                      maxStep: 10,
+                    },
+                    timeSelection: {
+                      start: {step: 0},
+                      end: {step: 5},
+                    },
+                  },
+                },
+              })
+            ),
+            'card1'
+          )
+        ).toBeUndefined();
+      });
+
+      it('returns undefined if no min max is defined', () => {
+        expect(
+          selectors.getMetricsCardTimeSelection(
+            appStateFromMetricsState(
+              buildMetricsState({
+                ...partialState,
+                cardStateMap: {
+                  card1: {
+                    timeSelection: {
+                      start: {step: 0},
+                      end: {step: 5},
+                    },
+                  },
+                },
+              })
+            ),
+            'card1'
+          )
+        ).toBeUndefined();
+      });
+
+      it('returns undefined when there is no card state', () => {
+        expect(
+          selectors.getMetricsCardTimeSelection(
+            appStateFromMetricsState(
+              buildMetricsState({
+                ...partialState,
+                cardStateMap: {},
+              })
+            ),
+            'card1'
+          )
+        ).toBeUndefined();
+      });
+
+      it('returns cards timeSelection if defined', () => {
```

Comment:
Personally I think this should be the first test in the set - to illustrate the happy path. And then all the other tests are places where the function fails to return a time selection or some interesting edge cases.